stubs/usersettings/usersettings.pyi

```diff
@@ -0,0 +1,14 @@
+from typing import Any, TypeVar
+
+_S = TypeVar("_S")
+
+class Settings(dict[str, object]):
+    app_id: str | None
+    settings_directory: str | None
+    settings_file: str | None
+    def __init__(self, app_id: str) -> None: ...
+    def add_setting(self, setting_name: str, setting_type: type[_S] = ..., default: _S | None = ...) -> None: ...
+    def load_settings(self) -> None: ...
+    def save_settings(self) -> None: ...
+    def __getattr__(self, setting_name: str) -> Any: ...
+    def __setattr__(self, setting_name: str, setting_value: object) -> None: ...
```

Comment:
More of an academic issue:

```suggestion
    def __setattr__(self, setting_name: str, setting_value: Any) -> None: ...
```

While at runtime every object is accepted, semantically it shouldn't be. E.g.:

```python
settings = Settings("")
settings.add_setting("x", int)
settings["x"] = ""  # works at runtime, but semantically wrong
```

Ideally, the type checker would complain about this, but there is no way we can express this. `Any` is a marker for that, while `object` really means "We accept everything here."